AdyenComponents/BACS Direct Debit/BACSDirectDebitComponent.swift

```diff
@@ -0,0 +1,101 @@
+
+    /// :nodoc:
+    public let apiContext: APIContext
+
+    /// :nodoc:
+    public let style: FormComponentStyle
+
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+
+    // MARK: - Initializers
+
+    /// Creates and returns a BACS Direct Debit component.
+    /// - Parameters:
+    ///   - paymentMethod: The BACS Direct Debit payment method.
+    ///   - apiContext: The API context.
+    ///   - style: The component's UI style.
+    ///   - localizationParameters: The localization parameters.
+    public init(paymentMethod: BACSDirectDebitPaymentMethod,
+                apiContext: APIContext,
+                style: FormComponentStyle = .init(),
+                localizationParameters: LocalizationParameters? = nil) {
+        self.paymentMethod = paymentMethod
+        self.apiContext = apiContext
+        self.style = style
+        self.localizationParameters = localizationParameters
+
+        let view = BACSDirectDebitInputFormViewController(title: paymentMethod.name,
+                                                          styleProvider: style)
+        self.viewController = view as UIViewController
+
+        let itemsFactory = BACSDirectDebitItemsFactory(styleProvider: style,
+                                                       localizationParameters: localizationParameters,
+                                                       scope: self)
+        let presenter = BACSDirectDebitPresenter(view: view,
+                                                 router: self,
+                                                 itemsFactory: itemsFactory)
+        view.presenter = presenter
+    }
+}
+
+// MARK: - BACSDirectDebitRouterProtocol
+
+extension BACSDirectDebitComponent: BACSDirectDebitRouterProtocol {
+
+    func presentConfirmation(with data: BACSDirectDebitData) {
+        // TODO: - Continue payment logic
+        // 1. Assamble confirmation scene
+        // 2. Present confirmation scene
+        print("PAYMENT: \(data)")
+
+        let confirmationView = UIViewController()
+        confirmationView.title = "Confirmation View"
+        confirmationView.view.backgroundColor = UIColor(red: 0.19, green: 0.84, blue: 0.78, alpha: 1.00)
+        viewController.navigationController?.pushViewController(confirmationView, animated: true)
+    }
+
+    func confirmPayment(with data: BACSDirectDebitData) {
+        // TODO: - Payment processing logic
+        guard let bacsDirectDebitPaymentMethod = paymentMethod as? BACSDirectDebitPaymentMethod else {
+            return
+        }
+        let bacsDirectDebitDetails = BACSDirectDebitDetails(paymentMethod: bacsDirectDebitPaymentMethod,
+                                                            holderName: data.holderName,
+                                                            bankAccountNumber: data.bankAccountNumber,
+                                                            bankLocationId: data.bankLocationId)
```

Comment:
you create the `bacsDirectDebitDetails ` and do nothing with it?

Reply:
The next PR will use it to submit the payment transaction.